From aa2e81201772db573c322d086533f587ef288bba Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 12 May 2020 19:39:13 +0300
Subject: [PATCH] [SW-0] ppboot: add version print

Since Ppboot has 2 Preboot versions, we would like to know what is
Ppboot version.
Add version print to Ppboot.

Change-Id: I3ca2e5e606a8e5c42956154a05e99769b80269eb
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/app/ppboot.c b/app/ppboot.c
index d45a216..dbf6c50 100644
--- a/app/ppboot.c
+++ b/app/ppboot.c
@@ -168,6 +168,11 @@
 
 __attribute__((weak)) void hl_init_pcie_phy_fw(void) {}
 
+static inline void ppboot_version(void)
+{
+	preboot_puts("Ppboot " VERSION_STR "\n");
+}
+
 void ppboot_main(void)
 {
 	u64 loadaddr;
@@ -184,6 +189,7 @@
 	preboot_sync_btl();
 	hl_uart_init();
 	ppboot_pcie_init();
+	ppboot_version();
 	rst_src_dump();
 
 	/* check if we are done here */
diff --git a/app/preboot.c b/app/preboot.c
index fc1666e..403dcea 100644
--- a/app/preboot.c
+++ b/app/preboot.c
@@ -5,7 +5,6 @@
  *
  */
 
-#include "auto_generated.h"
 #include <stddef.h>
 
 #include <psoc_global_conf_regs.h>
@@ -19,11 +18,6 @@
 #include "pll.h"
 #include "boot_if.h"
 #include "cmd_module.h"
-
-#define VERSION_MAX_LEN		128
-
-#define PREBOOT_VER_STR	"Preboot version " PREBOOT_VER \
-	" (" PREBOOT_DATE " - " PREBOOT_TIME ")"
 
 static struct cmd_module_ops cmd_ops[] = {
 	{	/* PCI module */
@@ -124,7 +118,7 @@
 }
 
 #ifndef TARGET_IS_TESTER
-static const char *preboot_ver_str = PREBOOT_VER_STR;
+static const char *preboot_ver_str = "Preboot " VERSION_STR;
 static char preboot_btl_ver_str[VERSION_MAX_LEN] = "BTL version ";
 
 __attribute__((weak)) void preboot_version(void)
diff --git a/include/preboot.h b/include/preboot.h
index 5ace7a0..6b56854 100644
--- a/include/preboot.h
+++ b/include/preboot.h
@@ -40,6 +40,13 @@
 
 #ifndef __ASSEMBLY__
 
+#include "auto_generated.h"
+
+#define VERSION_MAX_LEN		128
+
+#define VERSION_STR	"version " PREBOOT_VER \
+	" (" PREBOOT_DATE " - " PREBOOT_TIME ")"
+
 #define __packed	__attribute__((packed))
 
 typedef unsigned char u8;
